config/v1/techpreview.infrastructure.testsuite.yaml

```diff
@@ -12,3 +12,202 @@ tests:
       apiVersion: config.openshift.io/v1
       kind: Infrastructure
       spec: {}
+  onUpdate:
+  # platformStatus tests have to run in "onUpdate" because the Infrastructure has a
+  # status subresource, so we can't create an object with status.
+  - name: Should set load balancer type to OpenShiftManaged if not specified
+    initial: |
+      apiVersion: config.openshift.io/v1
+      kind: Infrastructure
+      spec:
+        platformSpec:
+          baremetal: {}
+          type: BareMetal
+    updated: |
+      apiVersion: config.openshift.io/v1
+      kind: Infrastructure
+      spec:
+        platformSpec:
+          baremetal: {}
+          type: BareMetal
+      status:
+        platform: BareMetal
+        platformStatus:
+          baremetal: {}
+          type: BareMetal
+    expected: |
+      apiVersion: config.openshift.io/v1
+      kind: Infrastructure
+      spec:
+        platformSpec:
+          baremetal: {}
+          type: BareMetal
+      status:
+        controlPlaneTopology: HighlyAvailable
+        infrastructureTopology: HighlyAvailable
+        platform: BareMetal
+        platformStatus:
+          baremetal:
+            loadBalancer:
+              type: OpenShiftManaged
+              openshiftManaged:
+                type: Default
```

Comment:
So for BGP we would write, e.g.:

```yaml
  loadBalancer:
    type: OpenShiftManaged
    openshiftManaged:
      type: BGP
      bgp:
      - subnet_id: xxxx-xxxx-xxxx-xxxx
        ASN: my_ASN
        peers:
        - ipv4: x.x.x.x
          ASN: peer_ASN
      - subnet_id: yyyy-yyyy-yyyy-yyyy
         ...
```

So we've got 2 discriminated unions here. Are we ok with this, or would we want to fold BGP into the outer union:

```yaml
  loadBalancer:
    type: OpenShiftManagedBGP
    openshiftManagedBGP:
    - subnet_id: xxxx-xxxx-xxxx-xxxx
      ASN: my_ASN
      peers:
      - ipv4: x.x.x.x
        ASN: peer_ASN
    - subnet_id: yyyy-yyyy-yyyy-yyyy
       ...
```

?

I'm ambivalent, btw, but I would like to make a deliberate choice based on a concrete use case.

Reply:
I thought with this API we would have something like:
```
loadBalancer:
  type: OpenShiftManaged
  openshiftManaged:
    type: BGP
    configs:
      - subnet_id: xxxx-xxxx-xxxx-xxxx
        ASN: my_ASN
        peers:
        - ipv4: x.x.x.x
          ASN: peer_ASN
      - subnet_id: yyyy-yyyy-yyyy-yyyy
...
```

I don't have any strong opinion around this neither so I'll let Joel / David to comment.

Reply:
I don't think we need to discriminate on the openshift managed differences, and would probably go for the latter
```
loadBalancer:
    type: OpenShiftManagedBGP
    openshiftManagedBGP:
    - subnet_id: xxxx-xxxx-xxxx-xxxx
      ASN: my_ASN
      peers:
      - ipv4: x.x.x.x
        ASN: peer_ASN
    - subnet_id: yyyy-yyyy-yyyy-yyyy
       ...
```
This means the union for now has no members, ie, just `type: UserManaged | OpenShiftManagedDefault`, but later we could expand this to include `OpenShiftManagedBGP` for platforms that need it, and then add a union member for the configuration

Reply:
So if I understand it right, you want me to remove `PlatformLoadBalancerOpenShiftManaged` type, and change `BareMetalPlatformLoadBalancer` to have `// +kubebuilder:validation:Enum:="OpenShiftManagedDefault";"UserManaged"` instead of `// +kubebuilder:validation:Enum:="OpenShiftManaged";"UserManaged"` ?

Reply:
cool. Thanks for the suggestion. I pushed the change.